youtube_dl/extractor/roosterteeth.py

```diff
@@ -0,0 +1,86 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    urlencode_postdata,
+)
+
+
+class RoosterTeethIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:(.+)\.)?roosterteeth\.com/episode/(?P<id>[^/?#&]+)'
```

Comment:
Do not capture group if you don't use it. Add `only_matching` tests for subdomains. Subdomain regex should not be greedy.